File: android_app/app/src/main/java/com/example/receiptApp/db/aggregate/Aggregate.kt
package com.example.receiptApp.db.aggregate

import android.location.Location
import android.net.Uri
import androidx.room.*
import com.example.receiptApp.db.tag.Tag
import java.util.*

/**
 * Aggregate
 *
 *
 *
 * @property id
 * @property tag_id
 * @property date
 * @property location
 * @property attachment
 * @property total_cost
 * @constructor Create empty Aggregate
 */

@Entity(
    tableName = "aggregate",
    indices = [Index("id")]
)
data class Aggregate @JvmOverloads constructor(
    @PrimaryKey(autoGenerate = true)
    var id: Long? = null,

    var tag_id: Long? = null,

    var date: Date? = null, // ESSENTIAL

    var location: Location? = null,

    var attachment: Uri? = null,

    var total_cost: Float = 0.0f,
){
    // fields not used in the table

    @Ignore
    var tag: String? = null
}

File: flutter_app/lib/Database/dataModels/aggregate.dart

class Aggregate{

  int? id;
  int? tag_id;
  final int date;
  String attachment;
  final double total_cost;
  String tag;

  Aggregate({
    this.id = null,
    required this.date,
    required this.total_cost,
    this.tag_id,
    this.attachment = "",
    this.tag = ""
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'tag_id': tag_id,
      'date': date,
      'attachment': attachment,
      'total_cost': total_cost
    };
  }

  static Aggregate fromMap(Map<String, dynamic> map){
    return Aggregate(
      id: map['id'],
      date: map['date'],
      total_cost: map['total_cost'],
      tag_id: map['tag_id'],
      attachment: map['attachment']
    );
  }

  @override
  String toString(){
    return 'Aggregate{id: $id, tag_id: $tag_id, date: $date, attachment: $attachment, total_cost: $total_cost }';
  }
}